File: SMP/SMP.Web/Modules/Sucursales/TblSucursalUber/TblSucursalUberRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace SMP.Sucursales;

[ConnectionKey("Default"), Module("Sucursales"), TableName("TBLSucursalUber")]
[DisplayName("Tbl Sucursal Uber"), InstanceName("Tbl Sucursal Uber")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
public sealed class TblSucursalUberRow : Row<TblSucursalUberRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Local Sap"), Column("LocalSAP"), Size(4), IdProperty, QuickSearch, NameProperty, ForeignKey("[dbo].[tblSucursal]", "LocalSap"), LeftJoin("jSucursales"), TextualField("NombreSuc")]
    public string LocalSap
    {
        get => fields.LocalSap[this];
        set => fields.LocalSap[this] = value;
    }
    [DisplayName("Nombre de Sucursal"), Expression("jSucursales.[NombreSuc]"), QuickSearch]
    public string NombreSucursal
    {
        get => fields.NombreSucursal[this];
        set => fields.NombreSucursal[this] = value;
    }
    public class RowFields : RowFieldsBase
    {
        public StringField LocalSap;
        public StringField NombreSucursal;

    }
}

File: SMP/SMP.Web/Modules/ServerTypes/Sucursales/TblSucursalUberRow.ts
import { fieldsProxy } from "@serenity-is/corelib/q";

export interface TblSucursalUberRow {
    LocalSap?: string;
    NombreSucursal?: string;
}

export abstract class TblSucursalUberRow {
    static readonly idProperty = 'LocalSap';
    static readonly nameProperty = 'LocalSap';
    static readonly localTextPrefix = 'Sucursales.TblSucursalUber';
    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<TblSucursalUberRow>();
}